joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp

```diff
@@ -104,6 +109,15 @@ class JointTrajectoryController : public controller_interface::ControllerInterfa
 
   bool is_halted = false;
 
+  typedef control_msgs::msg::JointTrajectoryControllerState JTrajControllerStateMsg;
+  typedef realtime_tools::RealtimePublisher<JTrajControllerStateMsg> StatePublisher;
+  typedef std::unique_ptr<StatePublisher> StatePublisherPtr;
+  rclcpp_lifecycle::LifecyclePublisher<JTrajControllerStateMsg>::SharedPtr publisher_;
+  StatePublisherPtr state_publisher_;
+
+  rclcpp::Duration state_publisher_period_ = rclcpp::Duration(1.0 / 50.0);
```

Comment:
Please add some tests where this param is set to 10, you bring up the controller and validate you get state feedback ~10Hz and one where it is set to 0 and you validate that there is nothing published on the state topic.

Reply:
Added test in [1cb140c](https://github.com/ros-controls/ros2_controllers/pull/25/commits/1cb140c3501bbb0b7ecad761ae4e53579132ad16)